smartsim/settings/pbsSettings.py

```diff
@@ -80,20 +91,35 @@ def __init__(
             time=time,
             **kwargs,
         )
-        self.resources = init_default({}, resources, dict)
+
         self._hosts: t.List[str] = []
 
+    @property
+    def resources(self):
+        return self._resources.copy()
+
+    @resources.setter
+    def resources(self, resources: dict[str, str | int]):
+        self._resources = resources.copy()
+        self._sanity_check_resources()
+
+
     def set_nodes(self, num_nodes: int) -> None:
         """Set the number of nodes for this batch job
 
-        If a select argument is provided in ``QsubBatchSettings.resources``
-        this value will be overridden
+        In PBS, 'select' is the more primitive way of describing how
+        many nodes to allocate for the job. 'nodes' is equivalent to
+        'select' with a 'place' statement. Assuming that only advanced
+        users would use 'set_resource' instead, defining the number of
+        nodes here is sets the 'nodes' resource.
 
         :param num_nodes: number of nodes
         :type num_nodes: int
         """
+
         if num_nodes:
-            self._nodes = int(num_nodes)
+            self.set_resource("nodes", num_nodes)
```

Comment:
I think casting `num_nodes` to `str` would make the resource dict easier to handle, as it would be of type `dict[str, t.Optional[str]]` instead of having to add `int` as another value type.